bigtop-packages/src/charm/layer-hive/lib/charms/layer/apache_hive.py

```diff
@@ -0,0 +1,137 @@
+import time
+import os
+import signal
+
+from charmhelpers.core import hookenv
+from charmhelpers.core import unitdata
+from charms import layer
+from jujubigdata import utils
+from charms.layer.apache_bigtop_base import Bigtop
+
+
+class Hive(object):
+    """
+    This class manages Hive.
+
+    :param DistConfig dist_config: The configuration container object needed.
+    """
+    def __init__(self, dist_config=None):
+        self.dist_config = dist_config or utils.DistConfig(data=layer.options('apache-bigtop-base'))
+
+    def install(self):
+        '''
+        Trigger the Bigtop puppet recipe that handles the Hive service.
+        '''
+        # Dirs are handled by the bigtop deb. No need to call out to
+        # dist_config to do that work.
+        self.dist_config.add_users()
+        roles = ['hive-client']
+
+        bigtop = Bigtop()
+        bigtop.render_site_yaml(roles=roles)
+        bigtop.trigger_puppet()
+
+    def initial_hive_config(self):
+        '''
+        Configure system-wide hive bits and get hive config files set up
+        so future config changes can be added where needed.
+        '''
+        # Configure system-wide bits
+        hive_bin = self.dist_config.path('hive') / 'bin'
+        with utils.environment_edit_in_place('/etc/environment') as env:
+            if hive_bin not in env['PATH']:
+                env['PATH'] = ':'.join([env['PATH'], hive_bin])
+            env['HIVE_CONF_DIR'] = self.dist_config.path('hive_conf')
+
+        # Copy templates to config files if they don't exist
+        hive_env = self.dist_config.path('hive_conf') / 'hive-env.sh'
+        hive_log4j = self.dist_config.path('hive_conf') / 'hive-log4j.properties'
+        hive_site = self.dist_config.path('hive_conf') / 'hive-site.xml'
+        if not hive_env.exists():
+            (self.dist_config.path('hive_conf') / 'hive-env.sh.template').copy(hive_env)
+        if not hive_site.exists():
```

Comment:
Overall this does not seem to be the bigtoppian way :)

Let me explain. Deploying hive bigtop copies under /etc/hive/conf/ the hive-site.xml found here:   https://github.com/juju-solutions/bigtop/blob/master/bigtop-deploy/puppet/modules/hadoop_hive/templates/hive-site.xml therefore this if statement and the entire attempt to write the default hive-site under etc/hive/config just fails. Later we parse this file to change the configuration. Why don't we patch the the template file provided by bigtop, make everything we need to change configurable and then change them via a puppet apply? Similarly if we need to changes to other files why don't we make the properties configurable via bigtop and push there the configuration?